spock-specs/src/test/groovy/org/spockframework/runtime/StoreSpec.groovy

```diff
@@ -0,0 +1,343 @@
+      'after SETUP - null',
+      'before FEATURE - data driven feature',
+      '	prev:     Stored at SETUP - null',
+      '	replaced: Stored at SETUP - null',
+      '	put:      Stored at FEATURE - data driven feature',
+      'after FEATURE - data driven feature',
+      'before CLEANUP - null',
+      '	prev:     Stored at FEATURE - data driven feature',
+      '	replaced: Stored at FEATURE - data driven feature',
+      '	put:      Stored at CLEANUP - null',
+      'after CLEANUP - null',
+      '# closing Stored at CLEANUP - null',
+      '# closing Stored at FEATURE - data driven feature',
+      '# closing Stored at SETUP - null',
+      '# closing Stored at ITERATION_EXECUTION - null',
+      '# closing Stored at INITIALIZER - null',
+      'after ITERATION_EXECUTION - null',
+      'before INITIALIZER - null',
+      '	prev:     Stored at FEATURE_EXECUTION - null',
+      '	replaced: null',
+      '	put:      Stored at INITIALIZER - null',
+      'after INITIALIZER - null',
+      'before ITERATION_EXECUTION - null',
+      '	prev:     Stored at INITIALIZER - null',
+      '	replaced: Stored at INITIALIZER - null',
+      '	put:      Stored at ITERATION_EXECUTION - null',
+      'before SETUP - null',
+      '	prev:     Stored at ITERATION_EXECUTION - null',
+      '	replaced: Stored at ITERATION_EXECUTION - null',
+      '	put:      Stored at SETUP - null',
+      'after SETUP - null',
+      'before FEATURE - data driven feature',
+      '	prev:     Stored at SETUP - null',
+      '	replaced: Stored at SETUP - null',
+      '	put:      Stored at FEATURE - data driven feature',
+      'after FEATURE - data driven feature',
+      'before CLEANUP - null',
+      '	prev:     Stored at FEATURE - data driven feature',
+      '	replaced: Stored at FEATURE - data driven feature',
+      '	put:      Stored at CLEANUP - null',
+      'after CLEANUP - null',
+      '# closing Stored at CLEANUP - null',
+      '# closing Stored at FEATURE - data driven feature',
+      '# closing Stored at SETUP - null',
+      '# closing Stored at ITERATION_EXECUTION - null',
+      '# closing Stored at INITIALIZER - null',
+      'after ITERATION_EXECUTION - null',
+      'after FEATURE_EXECUTION - null',
+      'before CLEANUP_SPEC - null',
+      '	prev:     Stored at FEATURE_EXECUTION - null',
+      '	replaced: Stored at FEATURE_EXECUTION - null',
+      '	put:      Stored at CLEANUP_SPEC - null',
+      'after CLEANUP_SPEC - null',
+      '# closing Stored at CLEANUP_SPEC - null',
+      '# closing Stored at FEATURE_EXECUTION - null',
+      '# closing Stored at SETUP_SPEC - null',
+      '# closing Stored at SPEC_EXECUTION - null',
+      '# closing Stored at SHARED_INITIALIZER - null',
+      'after SPEC_EXECUTION - null'
+    ]
+  }
+
+  @Shared
+  def supportedMethodKinds = ((MethodKind.values() as List) - [MethodKind.DATA_PROCESSOR, MethodKind.DATA_PROVIDER])
+
+  @Execution(ExecutionMode.SAME_THREAD)
+  def "store cleanup fails"() {
```

Comment:
Can we find a better name for the feature?
It does not test that the cleanup fails, but that cleanup failures do not prevent further cleanups to happen, doesn't it?